tests/notification.c

```diff
@@ -184,3 +188,72 @@ test_notification_bad_button (void)
 
   run_notification_test ("test5", notification_s, NULL, TRUE);
 }
+
+static void
+test_icon (const char *serialized_icon,
+           const char *exp_serialized_icon,
+           gboolean    exp_fail)
+{
+  g_autofree char *notification_s = NULL;
+  g_autofree char *exp_notification_s = NULL;
+
+  notification_s = g_strdup_printf ("{ 'title': <'test notification 7'>, "
+                                    "  'body': <'test notification body 7'>, "
+                                    "  'icon': <%s>, "
+                                    "  'default-action': <'test-action'> "
+                                    "}", serialized_icon);
+
+  if (exp_serialized_icon)
+    exp_notification_s = g_strdup_printf ("{ 'title': <'test notification 7'>, "
+                                          "  'body': <'test notification body 7'>, "
+                                          "  'icon': <%s>, "
+                                          "  'default-action': <'test-action'> "
+                                          "}", exp_serialized_icon);
+
+  run_notification_test ("test-icon", notification_s, exp_notification_s, exp_fail);
+}
+
+static void
+test_themed_icon ()
```

Comment:
All functions without arguments should be in the form of `foo (void)`. This commit contains a few of them, please update them all.